### ISSUE TITLE
Wrong DRVAL value not identified by S-58 check 1768

### ISSUE THREAD

**Issue by KlasOstergren-SMA** [iho-ohi/ENCWG-S58-ENC-Validation-Checks#47]
We have identified a case where incorrect DRVAL values (related to a sounding value) are not detected by any S-58 check. This is an effect of the changes that we made to check 1768 to allow isolated shoals with the same DRVAL1 and DRVAL2 values in ed.7. In the worst case, this can lead to an incorrect DRVAL1 value not being detected, affecting the safety contour.

In ed.6.1, all soundings with a depth value that was less than or equal to the DRVAL1 value were identified. 
![image](https://github.com/iho-ohi/ENCWG-S58-ENC-Validation-Checks/assets/81750471/d7ca80d3-e6e6-40a0-86ad-190b3a0ffb32)
In ed.7, soundings with a depth value that are equal to DRVAL1 in isolated shoals, are no longer detected.
![image](https://github.com/iho-ohi/ENCWG-S58-ENC-Validation-Checks/assets/81750471/1ac2a0fe-2e7a-4112-8701-3245d99600fe)

This is ok for soundings that are within a DEPARE where DRVAL1 and DRVAL2 are identical to the sounding. In this case, the VALDCO of the DEPCNT will also be the same. The problem occurs when the sounding has the same depth as DRVAL1, but DRVAL2 (and the VALDCO of the DEPCNT) has the value of the next deeper contour level used in the dataset.
The figure below shows four different cases where this is shown in red. In S-58 ed.6.1, this was detected by check 1768 but now it is not detected at all.
![image](https://github.com/iho-ohi/ENCWG-S58-ENC-Validation-Checks/assets/81750471/b9f42842-0e98-4ba2-8093-2294a0eee311)

To fix this, a new check (1768c?) needs to be added. I have made a first draft and would appreciate your comments.

`For each SOUNDG feature object where the depth value is equal to the DRVAL1 of the DEPARE feature object it is WITHIN AND DRVAL1 is not equal to DRVAL2 AND the depth value for the SOUNDG feature object is Equal to a VALDCO value on a DEPCNT feature object found in the ENC (if the DEPARE is an isolated shallow area)`

A 000 file containing the data for the figure above is attached.
[S58_check_1768.zip](https://github.com/iho-ohi/ENCWG-S58-ENC-Validation-Checks/files/14095277/S58_check_1768.zip)

**Comment by FrankHippmann** [comment]
> The problem occurs when the sounding has the same depth as DRVAL1, but DRVAL2 (and the VALDCO of the DEPCNT) has the value of the next deeper contour level used in the dataset.

Given the above statement, and the UOC guidance on isolated shoals, the correction for the red "Wrong encoding" example should be:
10m SOUNDG
DRVAL1 = 10
DRVAL2 = 15
VALDCO = 15

The problem here is that value of 10 for VALDCO is incorrect and it should be 15 (i.e. the same as the DRVAL2). I would not add a rather complicated check for SOUNDG when the problem is with the DEPCNT, not the SOUNDG.

Check 1771 already correctly identifies this incorrect DEPCNT.

![isolated shoal](https://github.com/user-attachments/assets/6a025cf2-ae27-4a73-895a-2f4044587156)

**Comment by KlasOstergren-SMA** [comment]
Frank is right that there is a difference between the dataset in this case and the description. An updated dataset is available here:  [S58_check_1768_new.zip](https://github.com/user-attachments/files/22084869/S58_check_1768_new.zip)

I can confirm that check 1771 identifies the depth contour, but the check message and the solution are somewhat misleading when they both refer to VALDCO for the depth contour being incorrect. In this case, it is DRVAL2 on the depth area that should be changed.

**Comment by FrankHippmann** [comment]
> I can confirm that check 1771 identifies the depth contour, but the check message and the solution are somewhat misleading when they both refer to VALDCO for the depth contour being incorrect. In this case, it is DRVAL2 on the depth area that should be changed.

Yes, the _Check message_ and _Check solution_ of 1771 assume that the issue is with the VALDCO, which may or may not be the case. We could update the _Check message_ and _Check solution_ to say that either the VALDCO or DRVAL2 are in error. Aside from that, I think that the actual _Check description_ of 1771 is correct.

**Comment by Richard-Fowle** [comment]
I propose amending the Check message to read :- 
Illogical value, of VALDCO of DEPCNT object or DRVAL2 of one of the adjacent DEPARE objects.

and Check solution to read :- 
Amend VALDCO or DRVAL2 to an appropriate value.

**Comment by Richard-Fowle** [comment]
Changes added to draft NE 9.0.0 of S-58
